### PR TITLE
Add Eartho Sign In Integration to StackAuth

### PR DESCRIPTION
**Pull request by earthodev**
### Description:
This PR adds support for [Eartho](https://eartho.io), a privacy-focused authentication provider that offers a secure alternative to traditional providers like Google and Facebook. Eartho enables data protection and user control, ideal for StackAuth users prioritizing privacy.

### Key Changes:
1. **Eartho Provider Support:**
   - Integrated Eartho as a provider in the OAuth system, added to `ProxiedOAuthProviderType`.
   - Added Eartho provider configuration in `oauth/providers/eartho.tsx` with custom styles, icon, and setup.

2. **Database & Schema Updates:**
   - Updated Prisma schema and seed files to support Eartho, enabling new and existing installations to include Eartho in authentication options.

3. **UI & Documentation:**
   - Custom Eartho OAuth button styling and icon added.
   - Documentation updates in `production.mdx` for setup and Eartho-specific configuration.

### Benefits:
- **Privacy-Centric**: Eartho prioritizes user data protection, preventing tracking.
- **Flexible Authentication**: Allows additional login options with enhanced control for users.
- **Easy Setup**: Preconfigured in seed files for new installations, with clear setup documentation.

Thank you for reviewing this PR – it expands secure, privacy-first authentication options for StackAuth users.

### CODE REVIEW

**Comment by N2D4** [comment]
Thanks! Could you please sign the CLA?

**Comment by dvir-daniel** [comment]
Done

**Comment by earthodev** [comment]
@N2D4 Could you please reset the jobs to run? We have authorized it. Thank you!

File: apps/backend/prisma/seed.ts [hunk 1]
```diff
@@ -311,4 +311,4 @@ seed().catch(async (e) => {
   await prisma.$disconnect();
   process.exit(1);
 // eslint-disable-next-line @typescript-eslint/no-misused-promises
-}).finally(async () => await prisma.$disconnect());
+}).finally(async () => await prisma.$disconnect());
```
Comment (N2D4) [inline apps/backend/prisma/seed.ts]:
```suggestion
}).finally(async () => await prisma.$disconnect());

```

File: packages/stack-shared/src/utils/oauth.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
-export const standardProviders = ["google", "github", "microsoft", "spotify", "facebook", "discord", "gitlab", "bitbucket", "linkedin", "apple", "x"] as const;
+export const standardProviders = ["google", "github", "microsoft", "eartho", "spotify", "facebook", "discord", "gitlab", "bitbucket", "linkedin", "apple", "x"] as const;
 // No more shared providers should be added except for special cases
-export const sharedProviders = ["google", "github", "microsoft", "spotify"] as const;
+export const sharedProviders = ["google", "github", "microsoft", "eartho"] as const;
```
Comment (N2D4) [inline packages/stack-shared/src/utils/oauth.tsx]: We try to keep the number of shared providers to a minimum, they're a headache to maintain
```suggestion
export const sharedProviders = ["google", "github", "microsoft", "spotify"] as const;
```

File: apps/backend/prisma/seed.ts (hunk 1 above)
Comment (dvir-daniel) [reply apps/backend/prisma/seed.ts]: Whats the change?

**Comment by N2D4** [comment]
Newline at the end of the file (it seems you accidentally removed it)

**Comment by N2D4** [comment]
Eartho is shutting down, so closing this

**Pull request closed without merging**